### ISSUE TITLE
5.0.4 -> 5.1.0 = Failed to set session cookie. Maybe you are using HTTP instead of HTTPS

### ISSUE THREAD

**Issue by johanvanhelden** [phpmyadmin/phpmyadmin#16819]
Version 5.0.4 was running just fine, no issues. After updating to 5.1.0 I was no longer able to login and keep getting the error:

`Failed to set session cookie. Maybe you are using HTTP instead of HTTP`

The domain is running on HTTPS.

I did not change anything inside the configuration file.

What I've tried to fix it myself:
- Cleared all my cookies and site data.
- Tried incognito mode.
- Installed a brand new browser

But the same issue occurs.

As soon as I roll back to the previous commit, with 5.0.4. everything is working again right away.

### Config
```php
<?php
/* vim: set expandtab sw=4 ts=4 sts=4: */
declare(strict_types=1);

$cfg['blowfish_secret'] = 'topsecret32charstringbutredactedforgithub';

$cfg['Servers'][1]['auth_type'] = 'cookie';
$cfg['Servers'][1]['host'] = 'localhost';
$cfg['Servers'][1]['compress'] = false;
$cfg['Servers'][1]['AllowNoPassword'] = false;

$cfg['UploadDir'] = '';
$cfg['SaveDir'] = '';
```

### Server:
```
nginx/1.15.8
Database client version: libmysql - mysqlnd 7.4.8
PHP extension: mysqli Documentation curl Documentation mbstring Documentation
PHP version: 7.4.8 (with PHP-FPM)
```

### Client configuration
```
- Browser: Firefox, Chrome & Brave
- Operating system: Windows 10
```

**Comment by williamdes** [comment]
> As soon as I roll back to the previous commit, with 5.0.4. everything is working again right away.

Hi @johanvanhelden 
Are you using a GIT setup ?

**Comment by johanvanhelden** [comment]
> 
> 
> > As soon as I roll back to the previous commit, with 5.0.4. everything is working again right away.
> 
> Hi @johanvanhelden
> Are you using a GIT setup ?

Yep!

**Comment by johanvanhelden** [comment]
Running yarn gives me the following error:
```
yarn run v1.22.4
$ yarn run css-compile --style=compressed && yarn run js-compile
$ sass themes/pmahomme/scss:themes/pmahomme/css themes/original/scss:themes/original/css themes/metro/scss:themes/metro/css themes/bootstrap/scss:themes/bootstrap/css --style=compressed
Error: Can't find stylesheet to import.
  ╷
3 │ @import "../../bootstrap/scss/bootstrap";
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  themes/pmahomme/scss/theme.scss 3:9      @import
  themes/pmahomme/scss/theme-rtl.scss 3:9  root stylesheet
```

But even if I try to submit the form when the styles are missing, I get the same ` Failed to set session cookie. Maybe you are using HTTP instead of HTTPS to access phpMyAdmin.`

**Comment by johanvanhelden** [comment]
> 
> 
> But did you run `yarn --production` before to install the packages ?
> 
> > Failed to set session cookie. Maybe you are using HTTP instead of HTTPS to access phpMyAdmin.
> 
> I would like to reproduce this issue, could you send your nginx configuration ?

I did not run `yarn --production` before because I rely on the already compiled CSS. 

I only exclude the vendor folder from GIT and install those myself after a deployment.

In regards to the nginx config, sure, I just took out some code in regards to SSL configuration and HTTP to HTTPS redirection:

```conf
// <SNIP> http -> https redirect

server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;
    server_name redacted.com;
    root /redacted/.;

    // <SNIP> (SSL cert config)

    add_header X-Frame-Options "SAMEORIGIN";
    add_header X-XSS-Protection "1; mode=block";
    add_header X-Content-Type-Options "nosniff";

    index index.html index.htm index.php;

    charset utf-8;

    auth_basic "Restricted";
    auth_basic_user_file location/to/.htpasswd;

    location / {
        try_files $uri $uri/ /index.php?$query_string;
    }

    location = /favicon.ico { access_log off; log_not_found off; }
    location = /robots.txt  { access_log off; log_not_found off; }

    access_log off;
    error_log  /var/log/nginx/redacted.com-error.log error;

    error_page 404 /index.php;

    location ~ \.php$ {
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass unix:/var/run/php/php7.4-fpm.sock;
        fastcgi_index index.php;
        include fastcgi_params;
    }

    location ~ /\.(?!well-known).* {
        deny all;
    }
}
```

Thanks for your time and help @williamdes !

**Comment by williamdes** [comment]
> I did not run yarn --production before because I rely on the already compiled CSS.

Oh okay, then this is more like a custom GIT install method :)

I had to add `fastcgi_param SCRIPT_FILENAME` probably because I am not using the same fpm config as you do

I did try without any issues the setup:
```nginx
# <SNIP> http -> https redirect

server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;
    server_name redacted.localhost;
    root "/mnt/Dev/@phpmyadmin/theREALphpMyAdminREPO";

    # <SNIP> (SSL cert config)
	ssl_certificate      /etc/nginx/nginx-selfsigned.crt;
	ssl_certificate_key  /etc/nginx/nginx-selfsigned.key;

    add_header X-Frame-Options "SAMEORIGIN";
    add_header X-XSS-Protection "1; mode=block";
    add_header X-Content-Type-Options "nosniff";

    index index.html index.htm index.php;

    charset utf-8;

    location / {
        try_files $uri $uri/ /index.php?$query_string;
    }

    location = /favicon.ico { access_log off; log_not_found off; }
    location = /robots.txt  { access_log off; log_not_found off; }

    access_log off;
    error_log  /var/log/nginx/redacted.localhost-error.log error;

    error_page 404 /index.php;

    location ~ \.php$ {
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass 8.1.williamdes.local:9000;
        fastcgi_index index.php;
	fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        include fastcgi_params;
    }

    location ~ /\.(?!well-known).* {
        deny all;
    }
}
```

**Comment by williamdes** [comment]
This issue is maybe related to: #15838

**Comment by johanvanhelden** [comment]
Upon further inspection I am noticing some console errors in the devtools that I am not getting on the 5.0.4 release, but not sure if that is related:

The resource from “https://redacted.com/js/vendor/jquery/jquery.ba-hashchange-2.0.js?v=5.1.0” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
dbruler.johanvanhelden.com
The resource from “https://redacted.com/js/vendor/jquery/jquery.debounce-1.0.6.js?v=5.1.0” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).

**Comment by johanvanhelden** [comment]
Upon further inspection I am noticing some console errors in the devtools that I am not getting on the 5.0.4 release, but not sure if that is related:

```
The resource from “https://redacted.com/js/vendor/jquery/jquery.ba-hashchange-2.0.js?v=5.1.0” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
The resource from “https://redacted.com/js/vendor/jquery/jquery.debounce-1.0.6.js?v=5.1.0” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
```

And if I try to visit those pages I am getting blank pages, there is no JS inside them.

And it is indeed a custom GIT setup. Because I need to add `config.inc.php` and `tmp/` to the gitignore in order to prevent the checkout from becoming dirty :-)

It's just a normal ZIP release download added to a repo.

**Comment by williamdes** [comment]
Another user reported the same issue: https://github.com/phpmyadmin/phpmyadmin/issues/15838#issuecomment-791906690

**Comment by williamdes** [comment]
Could you browse `https://redacted.com/js/vendor/jquery/jquery.ba-hashchange-2.0.js?v=5.1.0` and see what are the contents ?
Maybe also have a look to your webserver logs ?
I suspect `try_files $uri $uri/ /index.php?$query_string;` to create more problems than solutions

**Comment by williamdes** [comment]
PS: your domain leaked into the email notifications people receive, if it is highly sensitive you may need to take more securirty mesures :)
That said if you want me to have a look you can send me the link at williamdes @ wdes.fr

**Comment by johanvanhelden** [comment]
> 
> 
> Could you browse `https://redacted.com/js/vendor/jquery/jquery.ba-hashchange-2.0.js?v=5.1.0` and see what are the contents ?
> Maybe also have a look to your webserver logs ?
> I suspect `try_files $uri $uri/ /index.php?$query_string;` to create more problems than solutions

I think my problem is slightly different. When I visit those pages there is nothing inside them. No Javascript. They are blank pages.
Not seeing anything special in the logs either.

**Comment by johanvanhelden** [comment]
> 
> 
> PS: your domain leaked into the email notifications people receive, if it is highly sensitive you may need to take more security mesures :)
> That said if you want me to have a look you can send me the link at williamdes @ wdes.fr

Haha, nah, it's ok. It's already protected pretty well with an IP whitelist and htpasswd. But thanks!

**Comment by johanvanhelden** [comment]
You have to be kidding me. Found it. The fact that those files were showing blank pages bugged me. So I looked into it some more and noticed they were missing on the server.

Of course the `.gitignore` contains the `vendor` folder. But it did not only exclude the composer `vendor` folder but also the `/js/vendor` folder!

Some of those files `/js/vendor` have been tracked before during an earlier release so that is why things looked to be working fine, but the new files included in the 5.1.0 release have not been tracked. And it looks like those new js files are needed to login.

So the solution was to change the `.gitignore` `vendor` rule to `/vendor` and now the missing files from the `js/vendor` folder are tracked as well:

![image](https://user-images.githubusercontent.com/19389981/114608265-cb53d700-9c9d-11eb-87fd-9d5920f42c9e.png)

So everything is OK now. No problem on the PHPMyAdmin side.

Thanks for your support @williamdes !

**Comment by williamdes** [comment]
Glad to know you found the solution @johanvanhelden !
Enjoy your new version :)
